stubs/cffi/_cffi_backend.pyi

```diff
@@ -49,10 +50,86 @@ class CType:
     result: Incomplete
     def __dir__(self): ...
 
+@final
+class Lib:
+    def __delattr__(self, name): ...
+    def __dir__(self): ...
+    def __setattr__(self, name, value): ...
+
+@final
+class _CDataBase:
+    __name__: ClassVar[str] = ...
+    def __add__(self, other): ...
+    def __bool__(self): ...
+    def __call__(self, *args, **kwargs): ...
+    def __complex__(self): ...
+    def __delattr__(self, name): ...
+    def __delitem__(self, other): ...
+    def __dir__(self): ...
+    def __enter__(self): ...
+    def __eq__(self, other): ...
+    def __exit__(self, type, value, traceback): ...
+    def __float__(self): ...
+    def __ge__(self, other): ...
+    def __getitem__(self, index): ...
+    def __gt__(self, other): ...
+    def __hash__(self): ...
+    def __int__(self): ...
+    def __iter__(self): ...
+    def __le__(self, other): ...
+    def __len__(self): ...
+    def __lt__(self, other): ...
+    def __ne__(self, other): ...
+    def __radd__(self, other): ...
+    def __rsub__(self, other): ...
+    def __setattr__(self, name, value): ...
+    def __setitem__(self, index, object): ...
+    def __sub__(self, other): ...
+
+@final  # type: ignore[misc]
+class __CDataFromBuf(_CDataBase): ...
+
+@final  # type: ignore[misc]
+class __CDataGCP(_CDataBase):
+    def __del__(self, *args, **kwargs): ...
+
+@final  # type: ignore[misc]
+class __CDataOwn(_CDataBase):
+    def __delitem__(self, other): ...
+    def __getitem__(self, index): ...
+    def __len__(self): ...
+    def __setitem__(self, index, object): ...
+
+@final  # type: ignore[misc]
+class __CDataOwnGC(__CDataOwn): ...
+
+@final
+class __CData_iterator:
+    def __iter__(self): ...
+    def __next__(self): ...
+
+@final
+class __FFIGlobSupport: ...
+
+@final
+class buffer:
+    __hash__: ClassVar[None]  # type: ignore[assignment]
+    def __init__(self, *args, **kwargs) -> None: ...
+    def __delitem__(self, other): ...
+    def __eq__(self, other): ...
+    def __ge__(self, other): ...
+    def __getitem__(self, index): ...
+    def __gt__(self, other): ...
+    def __le__(self, other): ...
+    def __len__(self): ...
+    def __lt__(self, other): ...
+    def __ne__(self, other): ...
+    def __setitem__(self, index, object): ...
+
 class FFI:
     CData: TypeAlias = _CDataBase
     CType: TypeAlias = CType
```

Comment:
The pyright errors along the lines of `Type alias "CType" cannot use itself in its definition (reportGeneralTypeIssues)` are quite annoying (someone should probably file an issue about it). See https://github.com/python/typeshed/blob/42c044eb6772ee985fccc584f039998131bd5216/stdlib/multiprocessing/reduction.pyi#L55-L78 for examples of some hacky workarounds we use elsewhere.